src/ruby/yast/profiler.rb

```diff
@@ -0,0 +1,71 @@
+
+require "yast"
+
+module Yast
+  class Profiler
+    class << self
+      include Yast::Logger
+
+      RESULT_PATH = "/var/log/YaST2/profiler_result.txt"
+
+      # Start the Ruby Profiler. It start profilling. It also disables ruby VM
+      # optimizations, so code execution will be slower.
+      def start
+        raise "multiple profiller start detected" if @started
+        @original_compile_options = RubyVM::InstructionSequence.compile_option
+        @started = true
+        require "profiler"
+
+        # turn on tracing and turn off specialized instruction which replace
+        # some core ruby methods with its optimized variant to get complete
+        # profiling. More info in book "Ruby Under a Microscope: Learning Ruby
+        # Internals Through Experiment". Code is taken from ruby/lib/profile.rb
+        RubyVM::InstructionSequence.compile_option = {
+          trace_instruction:       true,
+          specialized_instruction: false
+        }
+
+        at_exit { stop }
+        Profiler__.start_profile
+      end
+
+      # stops profiling. If f is nil, then store profiling into RESULT_PATH file.
+      def stop(f = nil)
+        return File.open(RESULT_PATH, "w") { |f| stop(f) } unless f
+
+        Profiler__.print_profile(f)
+
+        RubyVM::InstructionSequence.compile_option = @original_compile_options
+        @started = false
+      end
+
+      # start the Ruby profiler if "Y2PROFILER" environment
+      # variable is set to "1" or "true"(the test is case
+      # insensitive, "y2profiler" variable can be also used)
+      def start_from_env
+        # do not evaluate again for each client started, run the evaluation only once
+        return if @profiler_handled
+        @profiler_handled = true
+
+        return unless env_value
+
+        log.info "profiler enabled"
+        start
+      end
+
+    private
+
+      # read the Y2PROFILER environment variable,
+      # do case insensitive match
+      # @return [Boolean] true if enabled
+      def env_value
+        # sort the keys to have a deterministic behavior and to prefer Y2PROFILER
+        # over the other variants, then do a case insensitive search
+        key = ENV.keys.sort.find { |k| k.match(/\AY2PROFILER\z/i) }
+        return false unless key
+
+        ["1", "true"].include?(ENV[key])
```

Comment:
I think we should allow only "1" to be consistent with the similar flags already used in YaST and Linuxrc. Otherwise also "yes" and "on" would be candidates and you would not remember what is allowed where.